docs/source/develop/keys.rst

```diff
@@ -0,0 +1,53 @@
+===============
+Key Pair Format
+===============
+
+Iroha uses key pairs (.pub and .priv keys) to sign transactions – every `account <../concepts_architecture/glossary.html#account>`_ has at least 1 pair.
+Some accounts (if `quorum <../concepts_architecture/glossary.html#quorum>`_ is more than 1) might have more `Signatories <../concepts_architecture/glossary.html#signatory>`_ that sign transactions – and each Signatory has a pair of keys.
+Cryptographic algorithms use those keys – and in Iroha we provide you with a choice – which algorithms to use.
+
+Supported Crypto Algorithms
+===========================
+
+Natively, HL Iroha uses a custom algorithm – SHA-3 ed25519.
+These keys are supported by all versions of Iroha, including the old ones.
+But as we all know, we need more universal options as well – that is why Iroha has `HL Ursa integration <../integrations/index.html#hyperledger-ursa>`_ – it is a library with different crypto algorithms, that allows to work with Iroha using more mainstream keys.
+Ursa provides Iroha with support of standard SHA-2 ed25519 algorithm.
+
+
+To provide easy solution that would allow using different algorithms without "breaking" backward compatibility, we introduced **multihash** format in Iroha.
+You can learn more about multihash `here <https://github.com/multiformats/multihash>`_.
+
+Generally, to create keys, different from the native SHA-3 ed25519 keys, you will need to bring them to this format:
+
+.. code-block:: shell
+
+	<varint hash function code><varint digest size in bytes><hash function output>
+
+
+.. note:: In multihash, varints are the Most Significant Bit unsigned varints (also called base-128 varints).
+
+
+Binary example from the multihash documentation (only 4 bytes for simplicity):
+
+.. code-block:: shell
+
+	fn code  dig size hash digest
+	-------- -------- ------------------------------------
+	00010001 00000100 101101100 11111000 01011100 10110101
+	sha1     4 bytes  4 byte sha1 digest
+
+
+If Iroha receives a standard public key of 32 bytes, it will treat is as a native Iroha key.
+If it receives a multihash public key, it will treat it based on the table below.
```

Comment:
I would add it to the table for illustrative clarity. like with empty first columns

Reply:
Kinda fixed?.. Could you please check the latest commit